lbrynet/dht/node.py

```diff
@@ -216,14 +216,13 @@ def getPeersForBlob(self, blob_hash):
         if result:
             if blob_hash in result:
                 for peer in result[blob_hash]:
-                    if self.node_id != peer[6:]:
-                        host = ".".join([str(ord(d)) for d in peer[:4]])
-                        if host == "127.0.0.1" and "from_peer" in result \
-                                and result["from_peer"] != "self":
-                            host = result["from_peer"]
-                        port, = struct.unpack('>H', peer[4:6])
-                        if (host, port) not in expanded_peers:
-                            expanded_peers.append((host, port))
+                    host = ".".join([str(ord(d)) for d in peer[:4]])
+                    if host == "127.0.0.1" and "from_peer" in result \
```

Comment:
what is this if statement doing here?

Reply:
I think it means: if this peer_list has been received from node X (where X != me) and I have '127.0.0.1' in the list, then substitute 127.0.0.1 address with X's address. Basically the code assumes that X had written 127.0.0.1 in the peer_list to represent itself, but to 'me' 127.0.0.1 does not make sense, therefore I use the real X's address.

To my understanding nobody is expected to store 127.0.0.1 in the DHT, therefore this should never happen. My guess is that this chunk was here to work around some bug.

does it make sense?

Reply:
Isn't all of `127.x.x.x` reserved for loopback (or other internal usage)? 

So wouldn't it make sense to prevent all 127 addresses from ever being stored in the DHT?

Reply:
Right. Theoretically all private addresses could be prevented from being stored in the DHT (127.0.0.0/8, 10.0.0.0/8, 172.16.0.0/12 and 192.168.0.0/16). Although there might be some cases where announcing private addresses may make sense, I believe it does not when nodes are connecting to the "global"/"public" DHT.

Therefore it should be safe to reject the store command and print some warning.

This is probably worth another patch.